### ISSUE TITLE
feat: Sigmoid activation layer

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#98]
## Summary

For binary classification tasks, the output layer needs a sigmoid activation to produce probabilities in [0, 1]. Currently available activations include ReLU, GELU, Softmax — but **Sigmoid is missing**.

## Proposed API

```go
// In package layers/activations
type Sigmoid[T numeric.Float] struct { ... }

func NewSigmoid[T numeric.Float](engine compute.Engine[T], ops numeric.Arithmetic[T]) *Sigmoid[T]

// Forward: σ(x) = 1 / (1 + exp(-x))
// Backward: σ(x) * (1 - σ(x))
```

## Use Case

Wolf's binary classification models predict price direction (up=1, down=0). The output layer currently produces unbounded values. Adding sigmoid on the output:
1. Constrains output to [0, 1] (interpretable as probability)
2. Compatible with BCE loss
3. Makes MLP output comparable to LightGBM probability output for ensemble blending
